docs/how_to_write_packages.md

```diff
@@ -1,3 +1,265 @@
+
+Then, you should also declare the _data-gatherers_ that your rules are going to need. For this example, let's just use `k8s/pods`.
+
+Finally, it's time to declare the rules for the policy. Rules are organized into sections. Every section has an ID, a name and a description. Also, every rule has its own ID, name and description. Additionally, rules can have additional metadata like a remediation or a set of related links.
+
+For simplicity's sake, this example contains just one section with one rule.
+
+```
+# preflight-packages/examples.jetstack.io/podsbestpractices/policy-manifest.yaml
+
+schema-version: "1.0.0"
+id: "podsbestpractices"
+namespace: "examples.jetstack.io"
+package-version: "1.0.0"
+root-query: "data.pods" # the concept of `root-query` is explained later in this doc
+data-gatherers:
+- k8s/pods
+sections:
+  - id: images
+    name: Images
+    description: "Restrictions over the images."
+    rules:
+      - id: tag_not_latest
+        name: "Tag is not latest"
+        description: >
+          Avoid using "latest" as tag for the image since.
+        remediation: >
+          Change your manifest and edit the Pod template so the image is pinned to a certain tag.
+        links:
+          - "https://kubernetes.io/docs/concepts/containers/images/"
+```
+
+## Writing the policy definition in REGO
+
+In the previous section, we created the _policy manifest_, which contains a human readable description of the rules in our policy. Now it's time to define the same rules in a language that is machine readable.
+
+### The REGO package
+
+Preflight relies on Open Policy Agent as the policy engine. REGO is OPA's language to define policies. You can find a comprenhensive [documentation](https://www.openpolicyagent.org/docs/latest/policy-language/).
+
+You can have multiple REGO files inside the directory of a Preflight package. One important think to take into consideration is that all the final definition of the rules must be in the same REGO package, and that package must be indicated in the _policy manifest_ using the `root-query` property.
+
+For instance, these snippet shows an arbitrary REGO rule in a package named `podsbestpractices`:
+
+```
+package pods
+
+import input["k8s/pods"] as pods
+
+preflight_tag_not_latest {
+  true
+}
+```
+
+As you can identify, the REGO package for that policy is `pods`. In this case, OPA's `root-query` is `data.pods`, and that is why in the previous section, `policy-manifest.yaml` contains `root-query: "data.pods"`.
+
+### Writing REGO rules
+
+REGO can be challenging at the beginning because it does not behaves like a traditional programming language. It is strongly recommended to read ["The Basics"](https://www.openpolicyagent.org/docs/latest/policy-language/#the-basics). Also, it is useful to have the language refence](https://www.openpolicyagent.org/docs/latest/policy-reference/) at a hand.
```

Comment:
```suggestion
REGO can be challenging at the beginning because it does not behaves like a traditional programming language. It is strongly recommended to read ["The Basics"](https://www.openpolicyagent.org/docs/latest/policy-language/#the-basics). Also, it is useful to have the [language reference](https://www.openpolicyagent.org/docs/latest/policy-reference/) at hand.
```